src/lambda/handler-runner/go-runner/GoRunner.js

```diff
@@ -0,0 +1,188 @@
+    this.#credentials = null
+    this.#handle = null
+  }
+
+  async run(event, context) {
+    const { dir } = pathParse(this.#handlerPath)
+    const handlerCodeRoot = dir.split(sep).slice(0, -1).join(sep)
+    const handlerCode = readFileSync(`${this.#handlerPath}.go`, 'utf8')
+    this.#tmpPath = resolve(handlerCodeRoot, 'tmp')
+    this.#tmpFile = resolve(this.#tmpPath, 'main.go')
+
+    const out = handlerCode.replace(
+      '"github.com/aws/aws-lambda-go/lambda"',
+      'lambda "github.com/icarus-sullivan/mock-lambda"',
+    )
+
+    try {
+      mkdirSync(this.#tmpPath, { recursive: true })
+    } catch (e) {
+      // @ignore
+    }
+
+    try {
+      writeFileSync(this.#tmpFile, out, 'utf8')
+    } catch (e) {
+      // @ignore
+    }
+
+    // Get go env to run this locally
+    if (!this.#goEnv) {
+      const goEnvResponse = spawnSync('go', ['env'], {
+        encoding: 'utf-8',
+      })
+
+      const goEnvString = goEnvResponse.stdout || goEnvResponse.stderr
+      this.#goEnv = goEnvString.split(EOL).reduce((a, b) => {
+        const [k, v] = b.split('="')
+        // eslint-disable-next-line no-param-reassign
+        a[k] = v ? v.slice(0, -1) : ''
+        return a
+      }, {})
+    }
+
+    // Get session credentials to pass to go
+    if (!this.#credentials) {
+      const sts = new STS()
+
+      const { Credentials } = await sts
+        .getSessionToken({
+          DurationSeconds: MIN_ALLOWED_SESSION_DURATION_S, // 900-inf
+        })
+        .promise()
+
+      this.#credentials = Credentials
+      this.#handle = setTimeout(
+        this.clearCredentials.bind(this),
+        SESSION_TIMEOUT_MS,
+      )
+    }
+
+    // Remove our root, since we want to invoke go relatively
+    const cwdPath = `${this.#tmpFile}`.replace(`${cwd()}${sep}`, '')
+    const cp = spawnSync(`go`, ['run', cwdPath], {
+      stdio: 'pipe',
+      env: {
+        ...this.#env,
+        ...this.#goEnv,
+        AWS_ACCESS_KEY_ID: this.#credentials.AccessKeyId,
+        AWS_SECRET_ACCESS_KEY: this.#credentials.SecretAccessKey,
+        AWS_SESSION_TOKEN: this.#credentials.SessionToken,
+        AWS_PROFILE: this.#profile,
+        AWS_LAMBDA_LOG_GROUP_NAME: context.logGroupName,
+        AWS_LAMBDA_LOG_STREAM_NAME: context.logStreamName,
+        AWS_LAMBDA_FUNCTION_NAME: context.functionName,
+        AWS_LAMBDA_FUNCTION_MEMORY_SIZE: context.memoryLimitInMB,
+        AWS_LAMBDA_FUNCTION_VERSION: context.functionVersion,
+        LAMBDA_EVENT: stringify(event),
+        LAMBDA_CONTEXT: stringify(context),
+        IS_LAMBDA_AUTHORIZER:
+          event.type === 'REQUEST' || event.type === 'TOKEN',
+        IS_LAMBDA_REQUEST_AUTHORIZER: event.type === 'REQUEST',
+        IS_LAMBDA_TOKEN_AUTHORIZER: event.type === 'TOKEN',
+        PATH: process.env.PATH,
+      },
+      encoding: 'utf-8',
+    })
+
+    if (cp.stderr) {
+      throw new Error(cp.stderr)
+    }
+
+    // Clean up after we created the temporary file
+    this.cleanup()
```

Comment:
I put this in to aggressively cleanup tmp files, but as a side-effect it might wipe sts session info on each call. One could argue that it is a good thing to only retain session credentials to the lambda invocation, but it might slowdown subsequent requests when obtaining new sessions. As a security concern I think it should be left in imo, but I'm open to changing it.